### ISSUE TITLE
about Feature Input

### ISSUE THREAD

**Issue by xuejian64** [Arcee-LYK/Multi-to-Single#2]
the code “eeg_X_train = np.array(eeg_X_train).transpose(1, 0, 2)” of data.py

Is the Seediv EEG feature used as published on the official website？
what is shape of eeg feature，is（sample_nums,channel,window_nums,5）?
thank you for  your outstanding work

**Comment by Arcee-LYK** [comment]
The data format on the official website should be [62, sample nums, 5]. Please convert it to [sample nums, 310(62*5)] format first, then manually add a window of length 5 to get data in [batch size, 5, 310] format.

**Comment by xuejian64** [comment]
Thank you for your reply.
I am reproducing the results of SEED-IV, but the results are not good. May I ask how many epochs you have run and details of some hyperparameters.

**Comment by Arcee-LYK** [comment]
Pre-training has a great impact on results. Please refer to the newly uploaded files ’pretrain_seediv.py‘ and ’new_data.py‘ to select the appropriate pre-training model.

**Comment by xxh-1998** [comment]
> Pre-training has a great impact on results. Please refer to the newly uploaded files ’pretrain_seediv.py‘ and ’new_data.py‘ to select the appropriate pre-training model.

Hi,

In the file pretrain_seediv.py, there's an import statement: from main_model import EE_VQVAE_Encoder.

Does this mean that a new model file named main_model.py is being used? I couldn't find main_model.py in the recent updates.

**Comment by Arcee-LYK** [comment]
Sorry some of the file names are not consistent, please change it to 'from M2S import Multi_to_Single'.

**Comment by xuejian64** [comment]
> 抱歉，部分文件名不一致，请将其更改为‘from M2S import Multi_to_Single’。

I have already tried it.
however,Class Multi_to_Single has not functions:single_modal_forward and same_eye_modal at the part of "eval_epoch"

**Comment by Arcee-LYK** [comment]
’M2S.py' has been updated, it should work now.

**Comment by xxh-1998** [comment]
> ’M2S.py' has been updated, it should work now.

Hi,

Thank you so much for sharing your work!

After reviewing the new pretraining code, my understanding is that for each session and each subject, you split the 24 film clips into the first 16 for training and the last 8 for validation. It seems this is done to find the best hyperparameters.

I have a couple of questions:

Is the data split in the same manner (i.e., first 16 clips for training, last 8 for validation per subject/session) during the finetuning phase as well?
Additionally, I was wondering if you have conducted any cross-subject or cross-session experiments. If so, could you share any insights into their performance?
Thank you again for your valuable contribution!

**Comment by Arcee-LYK** [comment]
> > ’M2S.py' has been updated, it should work now.
> 
> Hi,
> 
> Thank you so much for sharing your work!
> 
> After reviewing the new pretraining code, my understanding is that for each session and each subject, you split the 24 film clips into the first 16 for training and the last 8 for validation. It seems this is done to find the best hyperparameters.
> 
> I have a couple of questions:
> 
> Is the data split in the same manner (i.e., first 16 clips for training, last 8 for validation per subject/session) during the finetuning phase as well? Additionally, I was wondering if you have conducted any cross-subject or cross-session experiments. If so, could you share any insights into their performance? Thank you again for your valuable contribution!

1， Yes, your understanding is right and the data is split in the same way during fine-tuning.

2，We upload a new 'cross-subject' folder, which includes a simple cross-subject code added in the rebuttal stage. The cross-modal accuracy can basically reach above 0.8. Because the subject difference itself is an 'emotion-independent' factor, the use of the 'EI' module can solve the cross-subject problem to a certain extent, and the CLUB model itself is also used in some domain adaptation/generalization scenarios. For cross-session, my personal experience is that there is not much difference from the subject-dependent setting.

**Comment by xxh-1998** [comment]
> > > ’M2S.py' has been updated, it should work now.
> > 
> > 
> > Hi,
> > Thank you so much for sharing your work!
> > After reviewing the new pretraining code, my understanding is that for each session and each subject, you split the 24 film clips into the first 16 for training and the last 8 for validation. It seems this is done to find the best hyperparameters.
> > I have a couple of questions:
> > Is the data split in the same manner (i.e., first 16 clips for training, last 8 for validation per subject/session) during the finetuning phase as well? Additionally, I was wondering if you have conducted any cross-subject or cross-session experiments. If so, could you share any insights into their performance? Thank you again for your valuable contribution!
> 
> 1， Yes, your understanding is right and the data is split in the same way during fine-tuning.
> 
> 2，We upload a new 'cross-subject' folder, which includes a simple cross-subject code added in the rebuttal stage. The cross-modal accuracy can basically reach above 0.8. Because the subject difference itself is an 'emotion-independent' factor, the use of the 'EI' module can solve the cross-subject problem to a certain extent, and the CLUB model itself is also used in some domain adaptation/generalization scenarios. For cross-session, my personal experience is that there is not much difference from the subject-dependent setting.

Thank you for your patient response!
